libsolidity/interface/CompilerStack.cpp

```diff
@@ -173,6 +182,9 @@ bool CompilerStack::parse()
 		if (!resolver.registerDeclarations(*source->ast))
 			return false;
 
+	map<string, SourceUnit const*> sourceUnitsByName;
+	for (auto& source : m_sources)
```

Comment:
you mean by passing them as argument?  This would need to change again, when we introduce the importAST-function, which is an alternative step to parse....